scalafix-core/shared/src/main/scala/scalafix/internal/rule/NoInfer.scala

```diff
@@ -1,28 +1,44 @@
 package scalafix.internal.rule
 
 import scala.meta._
+import metaconfig.{Conf, Configured}
 import scalafix.lint.LintCategory
 import scalafix.lint.LintMessage
-import scalafix.rule.RuleCtx
+import scalafix.rule.{Rule, RuleCtx}
 import scalafix.rule.SemanticRule
 import scalafix.util.SemanticdbIndex
 import scalafix.util.SymbolMatcher
+import scalafix.internal.config.TargetSymbolsConfig
 
-case class NoInfer(index: SemanticdbIndex)
+final case class NoInfer(
+    index: SemanticdbIndex,
+    configuration: TargetSymbolsConfig)
     extends SemanticRule(index, "NoInfer")
     with Product {
-  private val badSymbol = SymbolMatcher.exact(NoInfer.badSymbols: _*)
-  val error: LintCategory =
+
+  private lazy val error: LintCategory =
     LintCategory.error(
       """The Scala compiler sometimes infers a too generic type such as Any.
         |If this is intended behavior, then the type should be explicitly type
         |annotated in the source.""".stripMargin
     )
+
+  private lazy val noInferSymbol: SymbolMatcher =
+    if (configuration.symbols.isEmpty)
+      SymbolMatcher.normalized(NoInfer.badSymbols: _*)
```

Comment:
I commented on SymbolMatcher equality in https://gitter.im/scalacenter/scalafix?at=59eb7f7401110b7231e44492